editoast/src/views/pagination.rs

```diff
@@ -121,29 +118,72 @@ pub trait PaginatedList: ListAndCount + 'static {
 
 impl<T> PaginatedList for T where T: ListAndCount + 'static {}
 
-#[derive(Debug, Clone, Copy, Deserialize, IntoParams)]
-#[into_params(parameter_in = Query)]
-pub struct PaginationQueryParams {
+#[derive(Debug, Clone, Copy, Deserialize)]
+pub struct PaginationQueryParams<const MAX_PAGE_SIZE: u64 = 25> {
     #[serde(default = "default_page")]
-    #[param(minimum = 1, default = 1)]
     pub page: u64,
-    #[param(minimum = 1, default = 25)]
     pub page_size: Option<u64>,
 }
 
+impl<const MAX_PAGE_SIZE: u64> utoipa::IntoParams for PaginationQueryParams<MAX_PAGE_SIZE> {
+    fn into_params(
+        _parameter_in_provider: impl Fn() -> Option<utoipa::openapi::path::ParameterIn>,
+    ) -> Vec<utoipa::openapi::path::Parameter> {
+        use serde_json::json;
+        use utoipa::openapi::KnownFormat;
+        use utoipa::openapi::ObjectBuilder;
+        use utoipa::openapi::Required;
+        use utoipa::openapi::SchemaFormat;
+        use utoipa::openapi::SchemaType;
+        use utoipa::openapi::path::ParameterBuilder;
+        use utoipa::openapi::path::ParameterIn;
+
+        [
+            ParameterBuilder::new()
+                .name("page")
+                .parameter_in(ParameterIn::Query)
+                .required(Required::False)
+                .schema(Some(
+                    ObjectBuilder::new()
+                        .schema_type(SchemaType::Integer)
+                        .format(Some(SchemaFormat::KnownFormat(KnownFormat::Int64)))
+                        .minimum(Some(1f64))
+                        .default(Some(json!(default_page()))),
+                ))
+                .build(),
+            ParameterBuilder::new()
+                .name("page_size")
+                .parameter_in(ParameterIn::Query)
+                .required(Required::False)
+                .schema(Some(
+                    ObjectBuilder::new()
+                        .schema_type(SchemaType::Integer)
+                        .format(Some(SchemaFormat::KnownFormat(KnownFormat::Int64)))
+                        .minimum(Some(1f64))
+                        .maximum(Some(MAX_PAGE_SIZE as f64))
+                        .default(Some(json!(MAX_PAGE_SIZE)))
+                        .nullable(true),
+                ))
+                .build(),
+        ]
+        .to_vec()
+    }
+}
+
 const fn default_page() -> u64 {
     1
 }
 
-impl PaginationQueryParams {
+impl<const MAX_PAGE_SIZE: u64> PaginationQueryParams<MAX_PAGE_SIZE> {
     /// Returns a pre-filled [SelectionSettings] from the pagination settings
     /// that can then be used to list or count models
     pub fn into_selection_settings<M: Model + 'static>(self) -> SelectionSettings<M> {
         self.into()
     }
 
-    pub fn validate(self, max_page_size: i64) -> Result<PaginationQueryParams> {
+    pub fn validate(self) -> Result<Self> {
         let (page, page_size) = self.unpack();
+        let max_page_size = MAX_PAGE_SIZE as i64;
```

Comment:
Does it make any sense that `page_size` ends up being an `i64`? I would expect a `u64` (and therefore avoid this fallible `as` conversion). This comment is tangential to the current PR so might be treated in another one.